src/pudl/transform/nrelatb.py

```diff
@@ -0,0 +1,368 @@
+        .sort_index()[["value"]]
+        .drop_duplicates()
+        .unstack(level="core_metric_parameter")
+    )
+    nrelatb_unstacked.columns = nrelatb_unstacked.columns.droplevel()
+    # this next step regarding the fcr param is only applicable to
+    # the table w/ the fcr param:
+    if "fcr" in nrelatb_unstacked:
+        nrelatb_unstacked = broadcast_fcr_param_across_tech_detail(
+            nrelatb_unstacked, table_unstacker.idx_unstacked
+        )
+    return nrelatb_unstacked
+
+
+class Unstacker(BaseModel):
+    """Class that defines how to unstack the :func:`transform_start` table into all of the tidy NREL tables."""
+
+    rate_table: TableUnstacker = TableUnstacker(
+        idx=[
+            "report_year",
+            "core_metric_case",
+            "core_metric_variable_year",
+            "cost_recovery_period_years",
+            "technology_description",
+            "core_metric_parameter",
+        ],
+        params=[
+            "inflation_rate",
+            "interest_during_construction_nominal",
+            "tax_rate_federal_and_state",
+            "interest_rate_calculated_real",
+            "interest_rate_nominal",
+            "rate_of_return_on_equity_nominal",
+            "rate_of_return_on_equity_calculated_real",
+        ],
+    )
+    scenario_table: TableUnstacker = TableUnstacker(
+        idx=rate_table.idx + ["scenario_atb"],
+        params=[
+            "debt_fraction",
+            "wacc_real",
+            "wacc_nominal",
+            "capital_recovery_factor",
+            "fuel_cost",
+            "fixed_charge_rate",
+        ],
+    )
+    tech_detail_table: TableUnstacker = TableUnstacker(
+        idx=scenario_table.idx
+        + [
+            "technology_description_detail_1",
+            "technology_description_detail_2",
+        ],
+        params=[
+            "capex",
+            "capacity_factor",
+            "opex_fixed",
+            "levelized_cost_of_energy",
+            "opex_variable",
+            # 2023 only params
+            "capex_overnight_additional",
+            "construction_finance_factor",
+            "grid_connection_cost",
+            "heat_rate",
+            "heat_rate_penalty",
+            "net_output_penalty",
+            "capex_overnight",
+        ],
+    )
+
+    @property
+    def params_all(self: Self) -> list[str]:
+        """Compilation of all of the parameter values from each of the tables.
+
+        Also check if there are no duplicate params. We expect all of the parameters
+        across the :class:`TableUnstacker` to be unique.
+        """
+        params_all = (
+            self.rate_table.params
+            + self.scenario_table.params
+            + self.tech_detail_table.params
+        )
+        assert not {x for x in params_all if params_all.count(x) > 1}
+        return params_all
+
+
+def broadcast_fcr_param_across_tech_detail(
+    nrelatb_unstacked: pd.DataFrame, idx_unstacked: list[str]
```

Comment:
now that this is only being used in `_core_nrelatb__yearly_projections_by_scenario`, i'd like to move this down in the module. but i'll keep it here for easier viewing of diffs/following comments for now.